skule_vote/frontend/ui/src/hooks/ElectionHooks.js

```diff
@@ -54,24 +55,45 @@ export const useGetEligibleElections = () => {
   };
 };
 
-export const useHandleSubmit = (electionId) => {
+export const useHandleSubmit = (setEligibleElections) => {
   const { enqueueSnackbar } = useSnackbar();
 
-  return function handleSubmit(ranking) {
-    // Get response here once API is set up
-    console.log(JSON.stringify({ electionId, ranking }, null, 2));
-
-    // Variant is repeated as a hack in order to use a custom snackbar
-    enqueueSnackbar(
-      { message: "Your vote has been successfully cast", variant: "success" },
-      { variant: "success" }
-    );
-    enqueueSnackbar(
-      {
-        message: "Unable with vote due to Error: <error from response>.",
-        variant: "error",
-      },
-      { variant: "error" }
-    );
+  return async function handleSubmit(electionId, ranking) {
+    if (electionId == null) {
+      return null;
+    }
+    try {
+      const response = await post("/api/vote/", { electionId, ranking });
+      if (statusIsGood(response.status)) {
+        enqueueSnackbar(
+          {
+            message: "Your vote has been successfully cast",
+            variant: "success",
+          },
+          { variant: "success" }
+        );
+        // Remove this election from the list of eligible elections
+        setEligibleElections((prevState) => {
+          if (prevState == null) {
+            return prevState;
+          }
+          const updated = { ...prevState };
+          updated[electionId] = null;
+          delete updated[electionId];
+          return updated;
+        });
```

Comment:
Instead of removing it on the frontend wouldn't the better way to do it is to request from the backend - that way they don't get out of sync

Reply:
I think it's fine. I can't think of a scenario where desync would happen with a 60s refresh frequency. Requesting all the elections again would be a waste of resources on the backend when the voter's behaviour is the only thing that can change this state